### PR TITLE
更新

### PR DESCRIPTION
**Pull request by oujunke**

### CODE REVIEW

**Comment by oujunke** [comment]
更新

File: Jint/Native/Global/GlobalObject.cs [hunk 1]
```diff
@@ -69,14 +70,28 @@ protected override void Initialize()
...
+        {
+            System.Threading.Tasks.Task.Run(() =>
+            {
+                if (arguments.Length == 2 || arguments[1].IsInteger())
+                {
+                    System.Threading.Thread.Sleep(arguments[1].AsInteger());
+                }
+                if (arguments[0] is ICallable callable)
+                {
+                    callable.Call(thisObj, Arguments.Empty);
```
Comment (ayende) [inline Jint/Native/Global/GlobalObject.cs:90]: This isn't thread safe.

File: Jint/Native/Global/GlobalObject.cs [hunk 2]
```diff
@@ -69,14 +70,28 @@ protected override void Initialize()
...
             SetProperties(properties);
         }
-
+        private JsValue SetTimeout(JsValue thisObj, JsValue[] arguments)
+        {
+            System.Threading.Tasks.Task.Run(() =>
+            {
+                if (arguments.Length == 2 || arguments[1].IsInteger())
+                {
+                    System.Threading.Thread.Sleep(arguments[1].AsInteger());
```
Comment (ayende) [inline Jint/Native/Global/GlobalObject.cs:86]: Use `Task.Delay` if you want this

File: Jint/Runtime/Interpreter/Statements/JintTryStatement.cs [hunk 3]
```diff
@@ -36,6 +36,7 @@ protected override Completion ExecuteInternal()
                 // execute catch
                 if (_catch != null)
                 {
+                    System.Console.WriteLine("Error:"+b.Value);
```
Comment (ayende) [inline Jint/Runtime/Interpreter/Statements/JintTryStatement.cs:39]: Remove this

File: Jint/Runtime/Descriptors/PropertyFlag.cs [hunk 4]
```diff
@@ -6,15 +6,37 @@ namespace Jint.Runtime.Descriptors
     public enum PropertyFlag
     {
         None = 0,
+        /// <summary>
+        /// 可数
```
Comment (ayende) [inline Jint/Runtime/Descriptors/PropertyFlag.cs:10]: Please use English for comments and names

**Review by ayende — COMMENTED** [review]
I'm going to assume that this isn't a real PR (given the comment) , but I gave it a quick run.

Note that implementing `setTimeout` is a _complex_ issue. You have to take into account that there may be concurrent tasks running (and the engine isn't thread safe).
You also have to account for the fact that the timeout may be after the current Jint run is over, in which case you may hold references to disposed objects.

**Comment by oujunke** [comment]
抱歉   我是个新手 操作失误

**Comment by oujunke** [comment]
Sorry, I am a novice, I made a mistake

**Pull request closed without merging**